src/hooks.c

```diff
@@ -0,0 +1,73 @@
+#include "hooks.h"
+
+unsigned int packet_hook(void *priv, struct sk_buff *skb, const struct nf_hook_state *state) {
+    struct nf_hook_params *params = (struct nf_hook_params *)priv;
+    Packet packet = to_packet(skb);
+    Rule *matched_rule;
+
+    if (packet.ip_header) {
+        matched_rule = get_matching_rule(&packet, params->rules, params->rule_count);
+        if (matched_rule) {
+            print_packet(&packet, KERN_INFO "Packet %pI4:%u to %pI4:%u matched a rule");
+            return handle_packet(&packet, matched_rule);
+        }
+    }
+
+    return NF_ACCEPT;
+}
+
+Rule post_routing_rules[1] = {
+    {
+        {
+            {}, 
+            {ntohl(0xC0A81101), htons(8000)},
+            IPPROTO_TCP
+        },
+        {
+            1, 
+            {
+                {ntohl(0xC0A81180)},
+                {ntohl(0xC0A81101), htons(8080)}
+            }
+        }
+    }
+};
+Rule pre_routing_rules[1] = {
```

Comment:
This part is now well documented.
I would start by explaining shortly what it does:
`// Changes the source port from 8080 to 8000`

Additional better document the struct:
```
// The filter matches any TCP packets originating from the specified source ip and port
{
    {ntohl(0xC0A81101), htons(8080)}, // Source IP: 192.168.17.1, Source Port: 8080
    {},                               // No destination IP or port specified (wildcard).
    IPPROTO_TCP                       // Protocol: TCP
}
```